### ISSUE TITLE
Align perf benchmark with vLLM benchmark

### ISSUE THREAD

**Issue by vkovacevicTT** [tenstorrent/tt-xla#5051]
This issue tracks the effort to transfer perf benchmark features to vLLM benchmark, since vLLM should be our main path. 

For simplicity, list of features will be broken into two parts: **Perf** and **Dev** features.

**Perf** features are used to increase benchmark LLMs perf and/or align them with requirements from where we get perf targets from, and are of higher priority than dev features.
List of perf features:
- [x] weight dtype ([PR](https://github.com/tenstorrent/tt-xla/pull/5056))
- [x] mixed precission ([PR](https://github.com/tenstorrent/tt-xla/pull/5056))
- [x] kv_cache dtype ([PR](https://github.com/tenstorrent/tt-xla/pull/5056))
- [x] optimizer level >= 1 (~~[failing](https://github.com/tenstorrent/tt-xla/issues/4990), tt-mlir [fix](https://github.com/tenstorrent/tt-mlir/pull/8678) merged~~, ~~[new issue](https://github.com/tenstorrent/tt-xla/issues/5227)~~ fixed, [PR](https://github.com/tenstorrent/tt-xla/pull/5410))
- [x] program trace
- [x] multichip mesh and sharding specs ([PR](https://github.com/tenstorrent/tt-xla/pull/4947))
- [x] batch_size=32 ([PR](https://github.com/tenstorrent/tt-xla/pull/5056) single-chip, multichip blocked by above [issue](https://github.com/tenstorrent/tt-xla/issues/5083))

**Dev** features are used for easier debugging, tracking model accuracy, regressions etc.
List of dev features:
- [ ] PCC check ([blocker](https://discuss.vllm.ai/t/how-to-obtain-the-logits-of-llm/847))
- [ ] accuracy (top-k) check
- [x] IR dump ([PR](https://github.com/tenstorrent/tt-xla/issues/5083))
- [ ] fusion check
- [ ] sharding metrics
- [ ] device perf
- [ ] num_layers

**Comment by vkovacevicTT** [comment]
cc @odjuricicTT @AleksKnezevic

**Comment by AleksKnezevic** [comment]
@mmanzoorTT FYI.

**Comment by odjuricicTT** [comment]
KV cache pcc check is something that we should look into adding as well

**Comment by vkovacevicTT** [comment]
batch_size=32 for LLMBOX models enabled in this PR: https://github.com/tenstorrent/tt-xla/pull/5159
Failures fixed in this PR: https://github.com/tenstorrent/tt-xla/issues/5207

All vllm multichip models are now running on batch_size=32 ✅

**Comment by vkovacevicTT** [comment]
opt-level=2 for all vllm models:
Issues fixed, working locally on llama-3.2-1b.
PR open, testing on all vllm models: https://github.com/tenstorrent/tt-xla/pull/5410
